### PR TITLE
Crit errors docs review

### PR DESCRIPTION
**Pull request by helenktsai**

### CODE REVIEW

File: nservicebus/hosting/critical-errors.md [hunk 1]
```diff
@@ -1,57 +1,63 @@
...
 
-snippet: StopEndpointInCriticalError
+1. Terminate the process. If using `Environment.FailFast` or `IHostApplicationLifetime.Stop`, the NServiceBus endpoint can attempt a graceful shutdown which can be useful in non-transactional processing environments.
+2. Ensure the environment is configured to automatically restart processes when they stop.
+  * IIS: The IIS host will automatically spawn a new instance.
+  * Windows Service: The OS can restart the service after 1 minute if [Windows Service Recovery](/nservicebus/hosting/windows-service.md#installation-setting-the-restart-recovery-options) is enabled.
+  * Docker: Ensure that containers are configured with `restart=always`. See [Start containers automatically (Docker.com)](https://docs.docker.com/config/containers/start-containers-automatically/)
 
-Note: [The Microsoft Generic Host's](/nservicebus/hosting/extensions-hosting.md) `IHostApplicationLifetime.Stop` method also stops the NServiceBus endpoint gracefully.
+## What if I don't like the default behavior?
```
Comment (tamararivera) [inline nservicebus/hosting/critical-errors.md:46]: Suggestion: most of the time is not a 'taste' issue, but more so a need
```suggestion
## What if I need to override the default behavior?
```

File: nservicebus/hosting/critical-errors.md [hunk 2]
```diff
@@ -1,57 +1,63 @@
...
+[Microsoft Generic Host's](/nservicebus/hosting/extensions-hosting.md) `IHostApplicationLifetime.Stop` method stops the NServiceBus endpoint gracefully.
 
-snippet: DefiningCustomHostErrorHandlingAction
+Alternatively, a call to `criticalErrorContext.Stop` can be used.
 
-### Stopping the endpoint
+snippet: StopEndpointInCriticalError
 
-When a critical error occurs, it is often unknown if the issue is recoverable. A sound strategy is to terminate the process (e.g. via `Environment.FailFast` or `IHostApplicationLifetime.Stop`) when a critical error occurs and rely on the process hosting environment to restart the process as a recovery mechanism. This is a resilient way to deal with critical errors.
+Warn: Calling `criticalErrorContext.Stop` without terminating the host process will only stop the NServiceBus endpoint without affecting the host process and other components running within the same process. Restarting the process after stopping the endpoint is recommended.
```
Comment (tamararivera) [inline nservicebus/hosting/critical-errors.md:36]: Nitpick: I feel it sounds less like a command this way 
```suggestion
Warn: Calling `criticalErrorContext.Stop` without terminating the host process will only stop the NServiceBus endpoint without affecting the host process and other components running within the same process. This is why restarting the process after stopping the endpoint is the recommended approach.
```

File: nservicebus/hosting/critical-errors.md [hunk 3]
```diff
@@ -1,57 +1,63 @@
...
 
-* IIS: The IIS host will automatically spawn a new instance.
-* Windows Service: The OS can restart the service after 1 minute if [Windows Service Recovery](/nservicebus/hosting/windows-service.md#installation-setting-the-restart-recovery-options) is enabled.
-* Docker: Ensure that containers are configured with `restart=always`. See [Start containers automatically (Docker.com)](https://docs.docker.com/config/containers/start-containers-automatically/)
+Provide a custom [action](https://learn.microsoft.com/en-us/dotnet/api/system.action-1) to override the default behavior:
+
+snippet: DefiningCustomHostErrorHandlingAction
 
-### A possible custom implementation
+### Example custom implementation
```
Comment (tamararivera) [inline nservicebus/hosting/critical-errors.md:60]: Nitpick:
```suggestion
### Example of a custom implementation
```

File: nservicebus/hosting/critical-errors.md [hunk 4]
```diff
@@ -1,57 +1,63 @@
...
 
-Whenever possible rely on the environment hosting the endpoint process to automatically restart it:
+* Sending a real-time notification to support personnel when the endpoint has raised a critical error.
+* Limiting the retries of the endpoint handler, e.g. when it might affect costs.
+* Automatically restarting the endpoint and resetting the transport connection to attempt to resolve underlying issues in receiving or dispatching messages.
 
-* IIS: The IIS host will automatically spawn a new instance.
-* Windows Service: The OS can restart the service after 1 minute if [Windows Service Recovery](/nservicebus/hosting/windows-service.md#installation-setting-the-restart-recovery-options) is enabled.
-* Docker: Ensure that containers are configured with `restart=always`. See [Start containers automatically (Docker.com)](https://docs.docker.com/config/containers/start-containers-automatically/)
+Provide a custom [action](https://learn.microsoft.com/en-us/dotnet/api/system.action-1) to override the default behavior:
```
Comment (tamararivera) [inline nservicebus/hosting/critical-errors.md:56]: Suggestion: More wordy words
```suggestion
To override the default behavior a custom [action](https://learn.microsoft.com/en-us/dotnet/api/system.action-1) needs to be provided:
```

File: nservicebus/hosting/critical-errors.md [hunk 5]
```diff
@@ -69,7 +75,7 @@ When implementing a custom critical error callback:
 
 ## Raising a critical error
 
-Any code in the endpoint can invoke the Critical Error action.
+Any code in the endpoint can invoke the `criticalError` action.
```
Comment (tamararivera) [inline nservicebus/hosting/critical-errors.md:78]: Nitpick: I don't like mentions to 'code'
```suggestion
Any part of the implementation of the endpoint can invoke the `criticalError` action.
```

**Review by awright18 — APPROVED** [review]

**Review by tamararivera — APPROVED** [review]

**Pull request merged**